test/testdevice.py

```diff
@@ -21,8 +21,10 @@ def has_audio_recording(self):
             self.devicename))
         return self.upload_audio_recording()
 
-    def upload_recording(self):
+    def upload_recording(self, properties = {}):
         props = self.get_new_recording_props()
+        for key in properties:
+            props[key] = properties[key]
```

Comment:
It looks like you just want to use the `update` dict method.

This would take care of the default argument handling and the property overrides:

```python
props = self.get_new_recording_props()
if properties:
    props.update(properties)
```